cmd/server/app/server.go

```diff
@@ -188,6 +188,7 @@ func getUDSListener(ctx context.Context, udsName string) (net.Listener, error) {
 	defer udsListenerLock.Unlock()
 	oldUmask := syscall.Umask(0007)
 	defer syscall.Umask(oldUmask)
+	os.Remove(udsName) // delete stale socket file, if it exists
```

Comment:
This should already be supported by passing the flag `--delete-existing-uds-file=true` ([code reference](https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/e162f71742aa5594f0aa12743ed9d7d8fb7c6bf7/cmd/server/app/server.go#L208))

Reply:
/hold

Reply:
Is there a reason we don't enable that flag by default?

Reply:
I suspect it is to avoid technically a backward incompatible change, but this has come up before (https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/443) and I would not be opposed since we're pre-1.0.

@cheftako WDYT?

Reply:
@jkh52 @tallclair  I plan create a new PR which removes the option and always deletes the stale socket file, if it exists. I see no good reason to have an option to keep this file. Is this ok for you?

Reply:
> new PR which removes the option and always delete ...

That would have the down side of removing a flag, which is itself a breaking change (for operators upgrading patch versions).

I propose we:

1. update the Kubernetes docs as @batistein suggests (valuable in any case for existing image versions). I can take this task.

2. Separately we can change the flag default. Would you adapt this PR to do that? It looks like @tallclair and I would both agree; @cheftako do you have any concern with it? (Justification: we are pre-1.0, and can't think of a reasonable use case for the false value behavior).

Reply:
Sent https://github.com/kubernetes/website/pull/39977